src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -262,6 +264,16 @@ public class ImportOrderCheck
      */
     private ImportOrderOption option = ImportOrderOption.UNDER;
 
+    /**
+     * Complied array of patterns for property {@code groups}.
+     */
+    private Pattern[] groupsReg = EMPTY_PATTERN_ARRAY;
+
+    /**
+     * Complied array of patterns for property {@code staticGroups}.
+     */
+    private Pattern[] staticGroupsReg = EMPTY_PATTERN_ARRAY;
+
```

Comment:
Add these two more fields to store the compiled regex expressions. I have to expose `String[]` to users, but we also need to store the complied regex (we don't want to compile every time we want to use the regex). I am curious whether there are better solutions to this, since this change brings us more suppressions of all kinds.

Reply:
to avoid dealing with long discussion on what is actually a type that user should care, I am ok with this update.
I hope in some future we will agree on type system and simplify this Check